course_flow/tasks.py

```diff
@@ -0,0 +1,314 @@
+            "b": workflow.code,
+            "c": _("Hours"),
+            "d": str(
+                workflow.time_general_hours + workflow.time_specific_hours
+            ),
+            "e": _("Time"),
+            "f": stringify(workflow.time_required)
+            + " "
+            + workflow.get_time_units_display(),
+        },
+        ignore_index=True,
+    )
+    df = df.append({"a": _("Ministerial Competencies")}, ignore_index=True)
+    df = df.append({"a": _("Competency"), "b": _("Title")}, ignore_index=True)
+    nodes = Node.objects.filter(linked_workflow=workflow).distinct()
+    parent_outcomes = []
+    for node in nodes:
+        outcomenodes = get_unique_outcomenodes(node)
+        parent_outcomes += OutcomeExportSerializer(
+            [ocn.outcome for ocn in outcomenodes], many=True
+        ).data
+    a = [get_str(outcome, "code") for outcome in parent_outcomes]
+    b = [get_str(outcome, "title") for outcome in parent_outcomes]
+    df = pd.concat([df, pd.DataFrame({"a": a, "b": b})])
+    if len(nodes) > 0:
+        df = df.append(
+            {
+                "a": _("Term"),
+                "b": WeekWorkflow.objects.get(week__nodes=nodes[0]).rank + 1,
+            },
+            ignore_index=True,
+        )
+        prereqs = Node.objects.filter(
+            outgoing_links__target_node__in=nodes
+        ).distinct()
+        postreqs = Node.objects.filter(
+            incoming_links__source_node__in=nodes
+        ).distinct()
+        if len(prereqs) > 0:
+            df = df.append(
+                {
+                    "a": _("Prerequisites"),
+                    "b": [get_displayed_title(req) for req in prereqs].join(
+                        ", "
+                    ),
+                },
+                ignore_index=True,
+            )
+        if len(postreqs) > 0:
+            df = df.append(
+                {
+                    "a": _("Required For"),
+                    "b": [get_displayed_title(req) for req in postreqs].join(
+                        ", "
+                    ),
+                },
+                ignore_index=True,
+            )
+    df = df.append(
+        {
+            "a": _("Course Outcome"),
+            "b": _("Sub-Outcomes"),
+            "c": _("Competencies"),
+            "d": _("Topics & Content"),
+            "e": _("Lessons/Activities"),
+            "f": _("Assessments"),
+        },
+        ignore_index=True,
+    )
+    for outcome in workflow.outcomes.all():
+        df = df.append(
+            get_framework_line_for_outcome(outcome), ignore_index=True
+        )
+    return df
+
+
+def get_workflow_outcomes_table(workflow):
+    outcomes = get_all_outcomes_ordered(workflow)
+    data = OutcomeExportSerializer(outcomes, many=True).data
+    df = pd.DataFrame(
+        data, columns=["code", "title", "description", "id", "depth"]
+    )
+    pd.set_option("display.max_colwidth", None)
+    return df
+
+
+@shared_task

```

Comment:
Check `try_async` decorator in dalite/celery.py.

Reply:
Just to confirm, when using `try_async` I should not manually use delay when calling the function, correct?